tests/integration/solidity_contracts/conftest.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+import pytest
+from starkware.starknet.core.os.contract_address.contract_address import (
+    calculate_contract_address_from_hash,
+)
+from starkware.starknet.testing.contract import StarknetContract
+from web3 import Web3
+
+from tests.utils.contracts import get_contract, use_kakarot_backend
+from tests.utils.helpers import hex_string_to_bytes_array
+from tests.utils.reporting import traceit
+
+logger = logging.getLogger()
+
+
+@pytest.fixture(scope="package")
+def get_starknet_address(account_proxy_class, kakarot):
```

Comment:
so the broad refactoring across conftest files are to ensure that fixtures only get loaded for the specific types of tests that depend on them?

 a brief description of the motivation  will help ensure that I'm on the same page with you and can provide a more informed review